custom_model_runner/datarobot_drum/drum/root_predictors/prediction_server.py

```diff
@@ -328,30 +331,34 @@ def _run_flask_app(self, app):
             processes = self._params.get("processes")
             logger.info("Number of webserver processes: %s", processes)
         try:
-            if RuntimeParameters.has("USE_NIM_WATCHDOG") and str(
-                RuntimeParameters.get("USE_NIM_WATCHDOG")
-            ).lower() in ["true", "1", "yes"]:
-                # Start the watchdog thread before running the app
-                self._server_watchdog = Thread(
-                    target=self.watchdog,
-                    args=(port,),
-                    daemon=True,
-                    name="NIM Sidecar Watchdog",
+            if self.flask_app:
+                # when running application via the command line (e.g., gunicorn worker)
+                pass
+            else:
+                if RuntimeParameters.has("USE_NIM_WATCHDOG") and str(
```

Comment:
this check for `has(<attribute>)` is redundant as you can check for `RuntimeParameters.get("USE_NIM_WATCHDOG", "no")` with default False  value.